samples/nrf9160/modem_shell/src/ppp/ppp_settings.c

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <strings.h>
+
+#include <assert.h>
+
+#include <shell/shell.h>
+#include <drivers/uart.h>
+#include <settings/settings.h>
+
+#include "ppp_ctrl.h"
+#include "ppp_settings.h"
+
+extern const struct shell *shell_global;
+
+#define PPP_SETT_KEY "ppp_settings"
+
+#define PPP_SETT_UART_BAUDRATE "uart_baudrate"
+#define PPP_SETT_UART_PARITY "uart_parity"
+#define PPP_SETT_UART_STOP_BITS "uart_stop_bits"
+#define PPP_SETT_UART_DATA_BITS "uart_data_bits"
+#define PPP_SETT_UART_FLOW_CTRL "uart_flow_ctrl"
+
+static const struct device *ppp_uart_dev;
+static bool ppp_uart_conf_valid;
+static struct uart_config ppp_uart_conf;
+
+/******************************************************************************/
+
+static int ppp_settings_uart_dev_conf_get(struct uart_config *uart_conf)
+{
+	int ret;
+
+	if (ppp_uart_dev) {
+		ret = uart_config_get(ppp_uart_dev, uart_conf);
+		if (ret != 0) {
+			shell_warn(shell_global, "Cannot get PPP uart config, ret %d", ret);
+		}
+	} else {
+		shell_warn(shell_global, "No PPP uart device with name %s",
+			   CONFIG_NET_PPP_UART_NAME);
+		ret = -1;
+	}
+	return ret;
+}
+
+static int ppp_settings_uart_dev_conf_set(struct uart_config *uart_conf)
+{
+	int ret;
+
+	if (ppp_uart_dev) {
+		ret = uart_configure(ppp_uart_dev, uart_conf);
+		if (ret != 0) {
+			shell_warn(shell_global, "Cannot get PPP uart config, ret %d", ret);
+		}
+	} else {
+		shell_warn(shell_global, "No PPP uart device with name %s",
+			   CONFIG_NET_PPP_UART_NAME);
+		ret = -1;
+	}
+	return ret;
+}
+
+/******************************************************************************/
+
+/**@brief Callback when settings_load() is called. */
+static int ppp_settings_handler(const char *key, size_t len, settings_read_cb read_cb, void *cb_arg)
+{
+	int ret = 0;
+
+	if (strcmp(key, PPP_SETT_UART_BAUDRATE) == 0) {
+		ret = read_cb(cb_arg, &ppp_uart_conf.baudrate, sizeof(ppp_uart_conf.baudrate));
+		if (ret < 0) {
+			shell_error(shell_global, "Failed to read uart_baudrate, error: %d", ret);
+			return ret;
+		}
+		return 0;
```

Comment:
These returns in each if branch are redundant.

Reply:
Changed.